job-info: misc cleanup

chu11 opened the pull request:
Misc cleanup, mostly coming from comments in #5467

Note that I just saw #5585 got submitted, dunno if there are conflicts ...

grondo commented:
> Note that I just saw https://github.com/flux-framework/flux-core/pull/5585 got submitted, dunno if there are conflicts ...

Unlikely. That one only touches the `kvs-watch` module.

garlick reviewed (approved):
Looks good, just one nit.

chu11 commented:
@garlick thanks!  Fixed that and I also noticed a line > 80 chars near that, so I added that in a dumb fix at the top of this commit series.

Setting MWP
